torch/csrc/autograd/profiler.cpp

```diff
@@ -162,9 +161,9 @@ void enableProfiler(ProfilerConfig config) {
         }
       },
       [](const RecordFunction& fn) {
-        if (fn.getThreadId() != 0) {
-          // If we've overridden the thread_id on the RecordFunction, then find
-          //  the eventList that was created for the original thread_id. Then,
+        if (fn.getThreadId() != RecordFunction::getCurrentThreadId()) {
```

Comment:
Thinking about this more, maybe `fn.getThreadId()` is better named `getCreatingThreadId` or `getStartCallbacksThreadId` to indicate that it returns the thread id where this record function was created. I can do the renaming in a follow up PR.

Reply:
sounds good, I'll update